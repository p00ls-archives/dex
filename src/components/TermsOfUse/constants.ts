export const ACCEPTED_TERMS_OF_USE_COOKIE = 'accepted_terms_of_use'
