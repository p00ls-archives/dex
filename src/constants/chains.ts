import ethereumLogoUrl from 'assets/images/ethereum-logo.png'

export enum SupportedChainId {
  MAINNET = 1,
  ROPSTEN = 3,
  RINKEBY = 4,
  GOERLI = 5,
  KOVAN = 42,
}

export const ALL_SUPPORTED_CHAIN_IDS: SupportedChainId[] = [
  SupportedChainId.MAINNET,
  SupportedChainId.ROPSTEN,
  SupportedChainId.RINKEBY,
  SupportedChainId.GOERLI,
  SupportedChainId.KOVAN,
]

export const L1_CHAIN_IDS = [
  SupportedChainId.MAINNET,
  SupportedChainId.ROPSTEN,
  SupportedChainId.RINKEBY,
  SupportedChainId.GOERLI,
  SupportedChainId.KOVAN,
] as const

export interface L1ChainInfo {
  readonly blockWaitMsBeforeWarning?: number
  readonly docs: string
  readonly explorer: string
  readonly infoLink: string
  readonly label: string
  readonly logoUrl?: string
  readonly rpcUrls?: string[]
  readonly nativeCurrency: {
    name: string // 'Goerli ETH',
    symbol: string // 'gorETH',
    decimals: number //18,
  }
}

export type ChainInfo = { readonly [chainId: number]: L1ChainInfo }

export const CHAIN_INFO: ChainInfo = {
  [SupportedChainId.MAINNET]: {
    docs: 'https://docs.uniswap.org/',
    explorer: 'https://etherscan.io/',
    infoLink: 'https://info.uniswap.org/#/',
    label: 'Ethereum',
    logoUrl: ethereumLogoUrl,
    nativeCurrency: { name: 'ETH', symbol: 'ETH', decimals: 18 },
  },
  [SupportedChainId.RINKEBY]: {
    docs: 'https://docs.uniswap.org/',
    explorer: 'https://rinkeby.etherscan.io/',
    infoLink: 'https://info.uniswap.org/#/',
    label: 'Rinkeby',
    nativeCurrency: { name: 'Rinkeby ETH', symbol: 'rinkETH', decimals: 18 },
  },
  [SupportedChainId.ROPSTEN]: {
    docs: 'https://docs.uniswap.org/',
    explorer: 'https://ropsten.etherscan.io/',
    infoLink: 'https://info.uniswap.org/#/',
    label: 'Ropsten',
    nativeCurrency: { name: 'Ropsten ETH', symbol: 'ropETH', decimals: 18 },
  },
  [SupportedChainId.KOVAN]: {
    docs: 'https://docs.uniswap.org/',
    explorer: 'https://kovan.etherscan.io/',
    infoLink: 'https://info.uniswap.org/#/',
    label: 'Kovan',
    nativeCurrency: { name: 'Kovan ETH', symbol: 'kovETH', decimals: 18 },
  },
  [SupportedChainId.GOERLI]: {
    docs: 'https://docs.uniswap.org/',
    explorer: 'https://goerli.etherscan.io/',
    infoLink: 'https://info.uniswap.org/#/',
    label: 'Görli',
    nativeCurrency: { name: 'Görli ETH', symbol: 'görETH', decimals: 18 },
  },
}
