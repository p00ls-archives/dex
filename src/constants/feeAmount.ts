import JSBI from 'jsbi'

export enum FeeAmount {
  LOWEST = 100,
  LOW = 500,
  MEDIUM = 5000,
  HIGH = 10000,
}

export const FEE_QUOTIENT = JSBI.BigInt(5) // =0.5%
