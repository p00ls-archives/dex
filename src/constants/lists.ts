const P00LS_LIST = 'https://tokens-assets.prod.p00ls.io/tokens.json'

export const UNSUPPORTED_LIST_URLS: string[] = []

// this is the default list of lists that are exposed to users
// lower index == higher priority for token import
const DEFAULT_LIST_OF_LISTS_TO_DISPLAY: string[] = [P00LS_LIST]

export const DEFAULT_LIST_OF_LISTS: string[] = [
  ...DEFAULT_LIST_OF_LISTS_TO_DISPLAY,
  ...UNSUPPORTED_LIST_URLS, // need to load dynamic unsupported tokens as well
]

// default lists to be 'active' aka searched across
export const DEFAULT_ACTIVE_LIST_URLS: string[] = [P00LS_LIST]
