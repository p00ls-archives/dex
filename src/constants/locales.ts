export const SUPPORTED_LOCALES = [
  // order as they appear in the language dropdown
  'en-US',
  'es-ES',
  'fr-FR',
  'pt-BR',
] as const
export type SupportedLocale = typeof SUPPORTED_LOCALES[number]

export const DEFAULT_LOCALE: SupportedLocale = 'en-US'

export { messages as DEFAULT_MESSAGES } from '../locales/en-US'

export const LOCALE_LABEL: { [locale in SupportedLocale]: string } = {
  'en-US': 'English',
  'es-ES': 'Español',
  'fr-FR': 'Français',
  'pt-BR': 'Português',
}
