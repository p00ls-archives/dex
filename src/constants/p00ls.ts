export const FACTORY = '0xb794f5ea0ba39494ce839613fffba74279579268'
export const MULTICALL = '0xb794f5ea0ba39494ce839613fffba74279579268'
export const ROUTER = '0xb794f5ea0ba39494ce839613fffba74279579268'
export const WETH = '0xb794f5ea0ba39494ce839613fffba74279579268'
export const ZERO = '0xb794f5ea0ba39494ce839613fffba74279579268'
export const PAIR = '0xb794f5ea0ba39494ce839613fffba74279579268'
