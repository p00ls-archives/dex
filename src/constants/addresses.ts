import { FACTORY_ADDRESS as V2_FACTORY_ADDRESS } from '@uniswap/v2-sdk'

import { constructSameAddressMap } from '../utils/constructSameAddressMap'
import { SupportedChainId } from './chains'

type AddressMap = { [chainId: number]: string }

export const UNI_ADDRESS: AddressMap = constructSameAddressMap('0x1f9840a85d5aF5bf1D1762F925BDADdC4201F984')
export const MULTICALL_ADDRESS: AddressMap = constructSameAddressMap('0x1F98415757620B543A52E61c46B32eB19261F984', [])

export const V2_FACTORY_ADDRESSES: AddressMap = constructSameAddressMap(V2_FACTORY_ADDRESS)
export const V2_ROUTER_ADDRESS: AddressMap = constructSameAddressMap('0x7a250d5630B4cF539739dF2C5dAcb4c659F2488D')

export const MERKLE_DISTRIBUTOR_ADDRESS: AddressMap = {
  [SupportedChainId.MAINNET]: '0x090D4613473dEE047c3f2706764f49E0821D256e',
}
export const ARGENT_WALLET_DETECTOR_ADDRESS: AddressMap = {
  [SupportedChainId.MAINNET]: '0xeca4B0bDBf7c55E9b7925919d03CbF8Dc82537E8',
}
export const ENS_REGISTRAR_ADDRESSES: AddressMap = {
  [SupportedChainId.MAINNET]: '0x00000000000C2E074eC69A0dFb2997BA6C7d2e1e',
  [SupportedChainId.ROPSTEN]: '0x00000000000C2E074eC69A0dFb2997BA6C7d2e1e',
  [SupportedChainId.GOERLI]: '0x00000000000C2E074eC69A0dFb2997BA6C7d2e1e',
  [SupportedChainId.RINKEBY]: '0x00000000000C2E074eC69A0dFb2997BA6C7d2e1e',
}
export const SOCKS_CONTROLLER_ADDRESSES: AddressMap = {
  [SupportedChainId.MAINNET]: '0x65770b5283117639760beA3F867b69b3697a91dd',
}
