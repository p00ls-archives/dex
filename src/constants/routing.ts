// a list of tokens by chain
import { Currency, Token } from '@uniswap/sdk-core'

import { SupportedChainId } from './chains'
import { ZEROZERO } from './tokens'

type ChainTokenList = {
  readonly [chainId: number]: Token[]
}

type ChainCurrencyList = {
  readonly [chainId: number]: Currency[]
}

// used to construct intermediary pairs for trading
export const BASES_TO_CHECK_TRADES_AGAINST: ChainTokenList = {
  [SupportedChainId.MAINNET]: [ZEROZERO[SupportedChainId.MAINNET]],
  [SupportedChainId.RINKEBY]: [ZEROZERO[SupportedChainId.RINKEBY]],
  [SupportedChainId.GOERLI]: [ZEROZERO[SupportedChainId.GOERLI]],
}
export const ADDITIONAL_BASES: { [chainId: number]: { [tokenAddress: string]: Token[] } } = {}
/**
 * Some tokens can only be swapped via certain pairs, so we override the list of bases that are considered for these
 * tokens.
 */
export const CUSTOM_BASES: { [chainId: number]: { [tokenAddress: string]: Token[] } } = {}

/**
 * Shows up in the currency select for swap and add liquidity
 */
export const COMMON_BASES: ChainCurrencyList = {
  [SupportedChainId.MAINNET]: [ZEROZERO[SupportedChainId.MAINNET]],
  [SupportedChainId.RINKEBY]: [ZEROZERO[SupportedChainId.RINKEBY]],
  [SupportedChainId.GOERLI]: [ZEROZERO[SupportedChainId.GOERLI]],
}

// used to construct the list of all pairs we consider by default in the frontend
export const BASES_TO_TRACK_LIQUIDITY_FOR: ChainTokenList = {}
export const PINNED_PAIRS: { readonly [chainId: number]: [Token, Token][] } = {}
