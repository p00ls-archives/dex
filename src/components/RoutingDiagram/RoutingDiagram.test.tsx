import { Currency, Percent, Token } from '@uniswap/sdk-core'
import { FeeAmount } from 'constants/feeAmount'
import { render } from 'test-utils'

import { SupportedChainId } from '../../constants/chains'
import RoutingDiagram, { RoutingDiagramEntry } from './RoutingDiagram'

const percent = (strings: TemplateStringsArray) => new Percent(parseInt(strings[0]), 100)

const USDC = new Token(SupportedChainId.MAINNET, '0x881ba05de1e78f549cc63a8f6cabb1d4ad32250d', 18, 'USDC', 'USDC')
const DAI = new Token(SupportedChainId.MAINNET, '0x4346cd535056593f184f655025cdb536b35b662f', 18, 'DAI', 'DAI')
const WBTC = new Token(SupportedChainId.MAINNET, '0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48', 18, 'WBTC', 'WBTC')

const singleRoute: RoutingDiagramEntry = { percent: percent`100`, path: [[USDC, DAI, FeeAmount.LOW]] }

const multiRoute: RoutingDiagramEntry[] = [
  { percent: percent`75`, path: [[USDC, DAI, FeeAmount.LOWEST]] },
  {
    percent: percent`25`,
    path: [
      [USDC, WBTC, FeeAmount.MEDIUM],
      [WBTC, DAI, FeeAmount.HIGH],
    ],
  },
]

jest.mock(
  'components/CurrencyLogo',
  () =>
    ({ currency }: { currency: Currency }) =>
      `CurrencyLogo currency=${currency.symbol}`
)

jest.mock(
  'components/DoubleLogo',
  () =>
    ({ currency0, currency1 }: { currency0: Currency; currency1: Currency }) =>
      `DoubleCurrencyLogo currency0=${currency0.symbol} currency1=${currency1.symbol}`
)

jest.mock('../Popover', () => () => 'Popover')

jest.mock('hooks/useTokenInfoFromActiveList', () => ({
  useTokenInfoFromActiveList: (currency: Currency) => currency,
}))

it('renders when no routes are provided', () => {
  const { asFragment } = render(<RoutingDiagram currencyIn={DAI} currencyOut={USDC} routes={[]} />)
  expect(asFragment()).toMatchSnapshot()
})

it('renders single route', () => {
  const { asFragment } = render(<RoutingDiagram currencyIn={USDC} currencyOut={DAI} routes={[singleRoute]} />)
  expect(asFragment()).toMatchSnapshot()
})

it('renders multi route', () => {
  const { asFragment } = render(<RoutingDiagram currencyIn={USDC} currencyOut={DAI} routes={multiRoute} />)
  expect(asFragment()).toMatchSnapshot()
})
