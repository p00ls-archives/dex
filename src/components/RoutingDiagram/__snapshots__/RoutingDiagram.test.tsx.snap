// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders multi route 1`] = `
<DocumentFragment>
  <div
    class="RoutingDiagram__Wrapper-sc-1sgyp7i-0 ityfBC css-vurnku"
  >
    <div
      class="sc-bdnxRM Row-sc-19fk4x0-0 RoutingDiagram__RouteContainerRow-sc-1sgyp7i-1 lmTMKd hTryuz ffflRm"
    >
      CurrencyLogo currency=USDC
      <div
        class="sc-bdnxRM Row-sc-19fk4x0-0 RoutingDiagram__RouteRow-sc-1sgyp7i-2 lmTMKd hTryuz epDqNq"
      >
        <div
          class="RoutingDiagram__DottedLine-sc-1sgyp7i-4 coPmVJ"
        >
          <svg
            class="RoutingDiagram__DotColor-sc-1sgyp7i-5 fjrkym"
          >
            dot_line.svg
          </svg>
        </div>
        <div
          class="Badge-sc-g51hsj-0 RoutingDiagram__OpaqueBadge-sc-1sgyp7i-6 bRVwmk eZDNFW"
        >
          <div
            class="theme__TextWrapper-sc-xx7zfj-0 jQATR css-15li2d9"
            style="word-break: normal;"
          >
            75%
          </div>
        </div>
        <div
          class="sc-bdnxRM Row-sc-19fk4x0-0 Row__AutoRow-sc-19fk4x0-3 iqvZFe hTryuz fRdFyI"
          style="justify-content: space-evenly; z-index: 2;"
          width="100%"
        >
          <div
            class="Badge-sc-g51hsj-0 RoutingDiagram__PoolBadge-sc-1sgyp7i-3 bRVwmk ZSmgP"
          >
            <div
              class="css-1t7xebc"
            >
              DoubleCurrencyLogo currency0=DAI currency1=USDC
            </div>
            <div
              class="theme__TextWrapper-sc-xx7zfj-0 jQATR css-15li2d9"
            >
              0.01%
            </div>
          </div>
        </div>
      </div>
      CurrencyLogo currency=DAI
    </div>
    <div
      class="sc-bdnxRM Row-sc-19fk4x0-0 RoutingDiagram__RouteContainerRow-sc-1sgyp7i-1 lmTMKd hTryuz ffflRm"
    >
      CurrencyLogo currency=USDC
      <div
        class="sc-bdnxRM Row-sc-19fk4x0-0 RoutingDiagram__RouteRow-sc-1sgyp7i-2 lmTMKd hTryuz epDqNq"
      >
        <div
          class="RoutingDiagram__DottedLine-sc-1sgyp7i-4 coPmVJ"
        >
          <svg
            class="RoutingDiagram__DotColor-sc-1sgyp7i-5 fjrkym"
          >
            dot_line.svg
          </svg>
        </div>
        <div
          class="Badge-sc-g51hsj-0 RoutingDiagram__OpaqueBadge-sc-1sgyp7i-6 bRVwmk eZDNFW"
        >
          <div
            class="theme__TextWrapper-sc-xx7zfj-0 jQATR css-15li2d9"
            style="word-break: normal;"
          >
            25%
          </div>
        </div>
        <div
          class="sc-bdnxRM Row-sc-19fk4x0-0 Row__AutoRow-sc-19fk4x0-3 iqvZFe hTryuz fRdFyI"
          style="justify-content: space-evenly; z-index: 2;"
          width="100%"
        >
          <div
            class="Badge-sc-g51hsj-0 RoutingDiagram__PoolBadge-sc-1sgyp7i-3 bRVwmk ZSmgP"
          >
            <div
              class="css-1t7xebc"
            >
              DoubleCurrencyLogo currency0=WBTC currency1=USDC
            </div>
            <div
              class="theme__TextWrapper-sc-xx7zfj-0 jQATR css-15li2d9"
            >
              0.3%
            </div>
          </div>
          <div
            class="Badge-sc-g51hsj-0 RoutingDiagram__PoolBadge-sc-1sgyp7i-3 bRVwmk ZSmgP"
          >
            <div
              class="css-1t7xebc"
            >
              DoubleCurrencyLogo currency0=DAI currency1=WBTC
            </div>
            <div
              class="theme__TextWrapper-sc-xx7zfj-0 jQATR css-15li2d9"
            >
              1%
            </div>
          </div>
        </div>
      </div>
      CurrencyLogo currency=DAI
    </div>
  </div>
</DocumentFragment>
`;

exports[`renders single route 1`] = `
<DocumentFragment>
  <div
    class="RoutingDiagram__Wrapper-sc-1sgyp7i-0 ityfBC css-vurnku"
  >
    <div
      class="sc-bdnxRM Row-sc-19fk4x0-0 RoutingDiagram__RouteContainerRow-sc-1sgyp7i-1 lmTMKd hTryuz ffflRm"
    >
      CurrencyLogo currency=USDC
      <div
        class="sc-bdnxRM Row-sc-19fk4x0-0 RoutingDiagram__RouteRow-sc-1sgyp7i-2 lmTMKd hTryuz epDqNq"
      >
        <div
          class="RoutingDiagram__DottedLine-sc-1sgyp7i-4 coPmVJ"
        >
          <svg
            class="RoutingDiagram__DotColor-sc-1sgyp7i-5 fjrkym"
          >
            dot_line.svg
          </svg>
        </div>
        <div
          class="Badge-sc-g51hsj-0 RoutingDiagram__OpaqueBadge-sc-1sgyp7i-6 bRVwmk eZDNFW"
        >
          <div
            class="theme__TextWrapper-sc-xx7zfj-0 jQATR css-15li2d9"
            style="word-break: normal;"
          >
            100%
          </div>
        </div>
        <div
          class="sc-bdnxRM Row-sc-19fk4x0-0 Row__AutoRow-sc-19fk4x0-3 iqvZFe hTryuz fRdFyI"
          style="justify-content: space-evenly; z-index: 2;"
          width="100%"
        >
          <div
            class="Badge-sc-g51hsj-0 RoutingDiagram__PoolBadge-sc-1sgyp7i-3 bRVwmk ZSmgP"
          >
            <div
              class="css-1t7xebc"
            >
              DoubleCurrencyLogo currency0=DAI currency1=USDC
            </div>
            <div
              class="theme__TextWrapper-sc-xx7zfj-0 jQATR css-15li2d9"
            >
              0.05%
            </div>
          </div>
        </div>
      </div>
      CurrencyLogo currency=DAI
    </div>
  </div>
</DocumentFragment>
`;

exports[`renders when no routes are provided 1`] = `
<DocumentFragment>
  <div
    class="RoutingDiagram__Wrapper-sc-1sgyp7i-0 ityfBC css-vurnku"
  />
</DocumentFragment>
`;
